### DISCUSSION TITLE
Does Budibase work as a headless CMS?  What are the API rate limits for the cloud hosted plans?

### DISCUSSION THREAD

**Discussion by sbutler-gh** [Budibase/budibase#12013] [Help] [answered]
I'm implementing a spreadsheet-like CMS backend for a project I'm working on.  The importance of a spreadsheet-like experience, is the need to enter news snippets and article headlines mutliple times a day -- giving a "feed" like experience.  A spreadsheet-like backend seems like the easiest way to manage this.

Would Budibase's API make it a suitable headless CMS for this?  I'm also open to approaches where updates to the backend could trigger static site builds on the Netlify deployment, in which case there wouldn't be API requests from clients to the Budibase backend.  If that would be pretty straightforward, it could also be a potential approach for this project.  So my main questions are:

1. Can Budibase be suitable as a headless CMS, with the cloud hosted plan?  What are the API rate limits?

2. Could Budibase be straightforward to implement a static site generator, which pulls data from a spreadsheet and triggers new static builds for a Netlify deployment?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @sbutler-gh,

You may really like the [grid block](https://docs.budibase.com/docs/grid-block) within Budibase, it lets you in-line edit whatever fields you want. So you could realistically do exactly what you are after multiple times a day.

1. I believe the rate limits to the api are 2 requests per second.
2. I'm not quite sure what Netlify deployments has to do with Budibase? Any new data added to your spreadsheets will automatically sync into the Budibase area. Currently I believe we only have compatibility with google sheets for spreadsheets.

I hope this helps.
